### ISSUE TITLE
Sketch 66.1 "Error: An Obj-C exception occurred."

### ISSUE THREAD

**Issue by MihaelN** [ingrammicro/puzzle-tokens#27]
I have tried importing demo .less files from the repository, and creating some of my own but I was stuck with the error that opened in plugin window.

"Error: An Obj-C exception occurred."

I am using macOS Mojave and the latest sketch 66.1

(I'll try to upload a screenshot later, firewall restrictions atm.)

**Comment by MaxBazarov** [comment]
Please do the following steps
1) Upgrade the plugin to 14.1.0 version
2) Enable Debug Logging in Configure Plugin settings
3) Reproduce an error
4) Open ~/Library/Logs/com.bohemiancoding.sketch3/Plugin\ Log.log. file
5) Show me last 50-100 lines
Thanks!

**Comment by andreasbrietzke** [comment]
I get the same error when trying the demo files and snippets from the Medium article.

Plugin is on 8.8.1, Show Debug Information ticked, but nothing in the log you mentioned.

Sketch 66.1, macOS Catalina (10.15.4)

**Comment by MaxBazarov** [comment]
> I get the same error when trying the demo files and snippets from the Medium article.
> Plugin is on 8.8.1, Show Debug Information ticked, but nothing in the log you mentioned

Sorry, but I wrote institutions for an another plugin. Puzzle Tokens has not debug logging yet.
I will do it ASAP.

"Show Debug Information" just shows internal JSON representation.

**Comment by MaxBazarov** [comment]
So, valid procedure is:

1. Upgrade the plugin to latest version (8.8.2)
2. Enable "Debug logging" in Configure settings
3. Reproduce an error
4. Open ~/Library/Logs/com.bohemiancoding.sketch3/Plugin\ Log.log. file
5. Attach log to this issue or send it to mbazarov@gmail.com.

Thanks!

<img width="635" alt="Screenshot 2020-06-04 at 15 38 19" src="https://user-images.githubusercontent.com/18418306/83757845-caafe480-a679-11ea-9c50-6fd5296533a4.png">

**Comment by andreasbrietzke** [comment]
This is what ends up in the log (tried with Buttons/Test.sketch and Buttons/Test-Blue.less):

```
run(): loadRules
/usr/local/bin/node/var/folders/yt/9cp5nzhs1ld56197tb29k38h0000gn/T/nsconvert_less.js -styles=/Users/andreasbrietzke/Downloads/puzzle-tokens-master/Styles/Tests/Buttons/Test-Blue.less -json=/var/folders/yt/9cp5nzhs1ld56197tb29k38h0000gn/T/nsdata.json
[ ERROR ] Error: An Obj-C exception occurred.
run(): loadRules: failed
```

**Comment by MaxBazarov** [comment]
@andreasbrietzke Thank you for cooperation. Could you run the following commands in Terminal and show me an output?
```
/usr/local/bin/node /var/folders/yt/9cp5nzhs1ld56197tb29k38h0000gn/T/nsconvert_less.js -styles=/Users/andreasbrietzke/Downloads/puzzle-tokens-master/Styles/Tests/Buttons/Test-Blue.less -json=/var/folders/yt/9cp5nzhs1ld56197tb29k38h0000gn/T/nsdata.json
cat /var/folders/yt/9cp5nzhs1ld56197tb29k38h0000gn/T/nsdata.json
```
Thanks! BTW, did you execute "npm i less -g" according to instructions?

**Comment by andreasbrietzke** [comment]
Yes, `less` and `sass` were both installed globally. But I figured what the problem is: 

I'm using `brew` to manage a couple packages and `npm` is among them. Apparently, `brew` does not symlink from `/usr/local/bin` to the actual installation location. 

Once I added the symlink manually, the plugin worked like a charm.

**Comment by MihaelN** [comment]
> Yes, `less` and `sass` were both installed globally. But I figured what the problem is:
> 
> I'm using `brew` to manage a couple packages and `npm` is among them. Apparently, `brew` does not symlink from `/usr/local/bin` to the actual installation location.
> 
> Once I added the symlink manually, the plugin worked like a charm.
> 
> (Thanks for the quick handling, though!)

Oh, that actually makes sense.  have also used brew to install `nvm`. I will give it a try today and will update if this fixes the problem for me too.

**Comment by MaxBazarov** [comment]
@andreasbrietzke  Thanks for information. 
I released 8.8.3. Now plugin will check /usr/local/bin/node existing and show a valid error message.
Moreover, I added ability to change node path in Settings.

I understand that it's not best solution. But it's better than nothing.
